octobot_evaluators/evaluator/abstract_evaluator.py

```diff
@@ -105,37 +115,61 @@ def get_is_time_frame_wildcard(cls) -> bool:
         """
         return True
 
-    def initialize(self):
-        set_tentacle_value(
-            matrix_id=self.matrix_id,
-            tentacle_type=self.get_eval_type(),
-            tentacle_value=None,
-            tentacle_path=get_matrix_default_value_path(
-                exchange_name=self.exchange_name,
-                tentacle_type=self.evaluator_type.value,
-                tentacle_name=self.get_name(),
-                cryptocurrency=self.cryptocurrency,
-                symbol=self.symbol,
-                time_frame=self.time_frame
-            )
-        )
+    def _get_tentacle_registration_topic(self, all_symbols_by_crypto_currencies, time_frames, real_time_time_frames):
+        currencies = [self.cryptocurrency]
+        symbols = [self.symbol]
+        available_time_frames = [self.time_frame]
+        if self.get_is_cryptocurrencies_wildcard():
+            currencies = all_symbols_by_crypto_currencies.keys()
+        if self.get_is_symbol_wildcard():
+            symbols = [currency_symbol
+                       for currency_symbols in all_symbols_by_crypto_currencies.values()
+                       for currency_symbol in currency_symbols]
+        if self.get_is_time_frame_wildcard():
+            available_time_frames = time_frames
+        return currencies, symbols, available_time_frames
+
+    def initialize(self, all_symbols_by_crypto_currencies, time_frames, real_time_time_frames):
+        currencies, symbols, time_frames = self._get_tentacle_registration_topic(all_symbols_by_crypto_currencies,
+                                                                                 time_frames,
+                                                                                 real_time_time_frames)
+        for currency in currencies:
+            for symbol in symbols:
```

Comment:
`[symbol for symbol in symbols if symbol is None or symbol in all_symbols_by_crypto_currencies[currency]]` and remove the next line, what do you think ?

Reply:
The problem with this syntax is that it creates an additional temporary list

Reply:
Sorry it's better with `symbol for symbol in symbols if symbol is None or symbol in all_symbols_by_crypto_currencies[currency]` as an iterator, you're right